### ISSUE TITLE
Snowflake won't sync without password (user using only private key)

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#30376]
### Describe the bug

If you create a user that is authenticated via private key, the sync will never happen: Metabase will connect to the DW, but the sync will never be able to finish as the connection pool is expecting a password to be created

### To Reproduce

1) create a user following this guide https://docs.snowflake.com/en/user-guide/key-pair-auth
2) connect to Metabase with that users
3) see the error

you'll see that Metabase connects, but the sync never finishes

### Expected behavior

We should use always the same authentication scheme once we connect for all processes

### Information about your Metabase installation

```JSON
- Metabase version: 1.46.1, but surely comes from 44
```

### Severity

P1

**Comment by john-metabase** [comment]
Small update, but not much new info: the same exact (as in `=`) connection spec map works for a non-pooled connection, but fails when passed in for use in a pooled connection (in `metabase.connection-pool` code). I'm trying to figure out why. Any ideas @camsaul, or @escherize?

**Comment by john-metabase** [comment]
Just to add a bit more info, after updating the code to properly retrieve the private key by `:private-key-id` instead of by `:private-key-value` on successive attempts to create a connection spec, we can connect and query using a non-pooled connection, but using the pooled version, e.g. during sync, yields the error:

```
ResourcePool could not acquire a resource from its primary factory or source.
        at com.mchange.v2.resourcepool.BasicResourcePool.awaitAvailable(BasicResourcePool.java:1507)
        at com.mchange.v2.resourcepool.BasicResourcePool.prelimCheckoutResource(BasicResourcePool.java:644)
        at com.mchange.v2.resourcepool.BasicResourcePool.checkoutResource(BasicResourcePool.java:554)
        at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutAndMarkConnectionInUse(C3P0PooledConnectionPool.java:758)
        at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutPooledConnection(C3P0PooledConnectionPool.java:685)
        ... 19 more
Caused by: net.snowflake.client.jdbc.SnowflakeSQLException: JWT token is invalid.
```

I don't think it's a thread safety issue. I am still trying to enable detailed logging from the Snowflake JDBC driver.

**Comment by john-metabase** [comment]
Narrowing in on it:

Snowflake driver logs show that the non-pooled connection logs in with `account=abcd1234` but the pooled version sends `account=abcd1234.us-east-2.aws`, which is likely the cause of the failure. Where this truncation happens, I have no idea... 🤷

**Comment by john-metabase** [comment]
It was not exactly the Snowflake driver, but the difference in `clojure.java.jdbc` handling of spec maps vs our `metabase.connection-pool` handling of the same map to create a connection, ending up with different connection properties being set in each case.

**Comment by likeshumidity** [comment]
Reopening as this is happening in 1.46.6. Not sure how since I tested in `master` post-merge and this issue was tagged with the 46.6 milestone. Having trouble with local build, so can't test on current `master` right now.

**Comment by john-metabase** [comment]
@darksciencebase @likeshumidity this should be resolved in 46 with the merge of #32076
